src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java

```diff
@@ -185,6 +204,36 @@ private static String getGoogModuleId(Node astRoot) {
             importRenameMap.put(localSymbolName, exportedSymbolName);
           }
         }
+      } else if (isVariableDestructuringAssignment(statement)) {
+        // `const B = goog.require()`
+        // `const {C, Clazz: RenamedClazz} = B`
+        // On separate lines
+        String destructuredVariable = statement.getFirstChild().getChildAtIndex(1).getString();
+
+        for (Node destructured : statement.getFirstFirstChild().children()) {
+          String originalName = destructured.getString();
+          // Destructuring can use the original name `const {A} = goog.require("foo.a")` or rename
+          // it `const {A: RenamedA} = ...`, and closure uses whichever in the symbol name it
+          // generates, so we have to extract it.
+          String variableName;
+          if (destructured.getFirstChild() != null) {
+            variableName = destructured.getFirstChild().getString();
+          } else {
+            variableName = originalName;
+          }
+
+          // The statements are iterated in order, so the statement with the goog.require() is already
+          // processed, so the value is already in the import rename map.
+          String exportedSymbolName =
+              importRenameMap.get(destructuredVariable) + "." + originalName;
+
+          importRenameMap.put(variableName, exportedSymbolName);
+          // If we're in a goog scope, there isn't a module id
```

Comment:
Feels like this check has to be moved higher up in the processing logic. I think (could be wrong) that one cannot have goog.require within a goog.scope.

The only two patterns I am familiar with is:
old
```
goog.require('a.b.c');
goog.scope(function() {
var c = a.b.c;
});
```
or new

```
goog.module(a.b.c);
const c = goog.require('a.b.c');
```

we are only trying to fix the new one, unless I forgot something.

Reply:
Goog.scope lets you write goog.module code inside a goog.provide file.  You can write:

```
goog.scope(function() {
var c = goog.require('a.b.c');
});
````

Which is what this is covering.